content/en/docs/internal-documentation/tutorials/adding_ado_alerts.md

```diff
@@ -0,0 +1,103 @@
+
+## Steps
+
+we can define rules for the addon-operator metrics in the [rhobs-rules-and-dashboards](https://gitlab.cee.redhat.com/service/rhobs-rules-and-dashboards)
+repo in the [rules/osd](https://gitlab.cee.redhat.com/service/rhobs-rules-and-dashboards/-/tree/main/rules/osd)
+folder (the tenants are added as specific folders) and tests for the prometheus rules are
+defined in the [tests/rules/osd](https://gitlab.cee.redhat.com/service/rhobs-rules-and-dashboards/-/tree/main/test/rules/osd)
+folder.
+The suggested way to add folder names and rules filenames for an alert is explained [here](https://gitlab.cee.redhat.com/service/rhobs-rules-and-dashboards/-/tree/main/rules#one-folder-per-targeted-rhobs-tenant).
+
+tldr: create file name with `.prometheusrule.yaml` as suffix at the end,
+add tenant label for the `PrometheusRule` object.
+
+### Adding promrules
+
+For creating a PoC alert we have selected `addon_operator_addon_health_info` metric
+since it basically explains the addon health for particular `version` and `cluster_id`.
+The metrics for creating alerts can be decided and the promql queries
+can be tested out from [promlens stage](https://promlens.stage.devshift.net)
+or [promlens prod](https://promlens.devshift.net).
+The metrics are scraped from addon-operator to
+observatorium-mst-stage (stage) / observatorium-mst-prod (prod) datasource.
+
+Sample `addon_operator_addon_health_info` metric data
+
+```yaml
+addon_operator_addon_health_info{_id="08d94ae0-a943-47ea-ac29-6cf65284aeba", container="metrics-relay-server", endpoint="https", instance="10.129.2.11:8443", job="addon-operator-metrics", name="managed-odh", namespace="openshift-addon-operator", pod="addon-operator-manager-7c9df45684-86mh4", prometheus="openshift-monitoring/k8s", receive="true", service="addon-operator-metrics", tenant_id="770c1124-6ae8-4324-a9d4-9ce08590094b", version="0.0.0"}
+```
+
+The particular metric gives information about
+`version`, `cluster_id (_id)` and `addon_name (name)` which can be used to create the alert.
+
+- We should ignore the metrics with version `"0.0.0"`
+- The addon health is `"Unhealthy"` if no version exists for the particular
+addon with [value 1](https://github.com/openshift/addon-operator#monitoring-and-metrics)
+- In theory, the latest version for the particular addon
+should have value of 1 if not then the addon is `"Unhealthy"`
+
+The prometheus rule for the addon_operator_addon_health_info metric is defined [here](https://gitlab.cee.redhat.com/service/rhobs-rules-and-dashboards/-/blob/main/rules/osd/addon-operator-addon-health-info-rules.prometheusrule.yaml)
+
+```yaml
+expr: (count by (name, _id) (addon_operator_addon_health_info{version!="0.0.0"})) - (count by (name,_id) (addon_operator_addon_health_info{version!="0.0.0"} == 0)) == 0
```

Comment:
Is this expression the same as
```
(count by (name,_id) (addon_operator_addon_health_info{version!="0.0.0"} != 0)) == 0
```

Reply:
so the promql won't return anything if `addon_operator_addon_health_info{version!="0.0.0"} != 0` doesn't return any hits. so subtracting 0 value metrics from everything is the way

Reply:
I tested this out, in the prom tests

Reply:
Ohhh yeah I understand :) Okay cool!